### ISSUE TITLE
Problem with Unicode

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1533]
Hello.

I'm trying to use inheritance_diagram module for classes with russian docstrings and I get Unicode error:

```
stas@stas-Aspire-Z3620:~/workspace/test_sphinx$ cat /tmp/sphinx-err-byr6rf.log
# Sphinx version: 1.2.2
# Python version: 2.7.5+
# Docutils version: 0.12 release
# Jinja2 version: 2.7.3
# Loaded extensions:
#   sphinx.ext.graphviz from /usr/local/lib/python2.7/dist-packages/sphinx/ext/graphviz.pyc
#   sphinx.ext.autodoc from /usr/local/lib/python2.7/dist-packages/sphinx/ext/autodoc.pyc
#   sphinx.ext.intersphinx from /usr/local/lib/python2.7/dist-packages/sphinx/ext/intersphinx.pyc
#   sphinx.ext.inheritance_diagram from /usr/local/lib/python2.7/dist-packages/sphinx/ext/inheritance_diagram.pyc
#   sphinx.ext.viewcode from /usr/local/lib/python2.7/dist-packages/sphinx/ext/viewcode.pyc
#   sphinx.ext.todo from /usr/local/lib/python2.7/dist-packages/sphinx/ext/todo.pyc
#   sphinx.ext.oldcmarkup from /usr/local/lib/python2.7/dist-packages/sphinx/ext/oldcmarkup.pyc
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/sphinx/cmdline.py", line 254, in main
    app.build(force_all, filenames)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/application.py", line 212, in build
    self.builder.build_update()
  File "/usr/local/lib/python2.7/dist-packages/sphinx/builders/__init__.py", line 214, in build_update
    'out of date' &#37; len(to_build))
  File "/usr/local/lib/python2.7/dist-packages/sphinx/builders/__init__.py", line 276, in build
    self.write(docnames, list(updated_docnames), method)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/builders/__init__.py", line 320, in write
    self._write_serial(sorted(docnames), warnings)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/builders/__init__.py", line 333, in _write_serial
    self.write_doc(docname, doctree)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/builders/html.py", line 433, in write_doc
    self.docwriter.write(doctree, destination)
  File "/usr/local/lib/python2.7/dist-packages/docutils/writers/__init__.py", line 80, in write
    self.translate()
  File "/usr/local/lib/python2.7/dist-packages/sphinx/writers/html.py", line 51, in translate
    self.document.walkabout(visitor)
  File "/usr/local/lib/python2.7/dist-packages/docutils/nodes.py", line 174, in walkabout
    if child.walkabout(visitor):
  File "/usr/local/lib/python2.7/dist-packages/docutils/nodes.py", line 174, in walkabout
    if child.walkabout(visitor):
  File "/usr/local/lib/python2.7/dist-packages/docutils/nodes.py", line 166, in walkabout
    visitor.dispatch_visit(self)
  File "/usr/local/lib/python2.7/dist-packages/docutils/nodes.py", line 1882, in dispatch_visit
    return method(node)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/ext/inheritance_diagram.py", line 352, in html_visit_inheritance_diagram
    alt='Inheritance diagram of ' + node['content'])
  File "/usr/local/lib/python2.7/dist-packages/sphinx/ext/graphviz.py", line 194, in render_dot_html
    fname, outfn = render_dot(self, code, options, format, prefix)
  File "/usr/local/lib/python2.7/dist-packages/sphinx/ext/graphviz.py", line 126, in render_dot
    ).encode('utf-8')
UnicodeDecodeError: 'ascii' codec can't decode byte 0xd0 in position 198: ordinal not in range(128)

```

For example I use such file:

```
#!python

#-*- coding: utf-8 -*-
"""
Тестовая документация

.. inheritance-diagram:: main

"""


class A:

    "Класс 1"
    pass


class B(A):

    "Класс 2"
    pass


class C(A):

    "Класс 3"
    pass
```

If I use English docstrings everything is OK. 
I suggest such silution:

in file sphinx/ext/graphviz.py in function render_dot(line 145) don't use encode, because expression in brackets is str(not unicode).

python version 2.7
Sphinx version 1.2.2

Full test project is attached.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1533
- Originally reported by: Anonymous
- Originally created at: 2014-08-14T11:23:42.169

**Comment by shimizukawa** [comment]
_From Стас Каледин on 2014-08-14 09:24:34+00:00_

Sorry, I forgot to log in. It's my issue

**Comment by shimizukawa** [comment]
_From WAKAYAMA shirou on 2014-08-14 11:38:54+00:00_

I made a https://bitbucket.org/birkenfeld/sphinx/pull-request/270. I confirmed this resolves your test_sphinx.zip.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-08-14 15:33:28+00:00_
- importing the https://bitbucket.org/birkenfeld/sphinx/pull-request/270 change to stable: Non-ASCII docstring cause UnicodeDecodeError when uses with inheritance-diagram directive. Thanks to WAKAYAMA shirou. Closes #1533

→ <<cset b5f351064c8b51672fcae7b9aa6e5634338cc593>>
